### ISSUE TITLE
docs: update CLI documentation for distil serve and new unsloth flags

### ISSUE THREAD

**Issue by egesabanci** [egesabanci/sped#83]
## Problem

The CLI documentation (`docs/cli.md`) has not been updated to reflect the Unsloth backend integration (PRs #63–#66) and the new flags added in the working tree:

### Missing from docs:
1. `sped serve --backend unsloth` — Unsloth 4-bit inference backend
2. `sped distil run --backend unsloth` — Unsloth-native fine-tuning
3. `sped distil run --draft-dtype bf16|4bit` — draft model precision
4. 4-bit model load caching behavior (creates `*-4bit-cache` directories)
5. `sped distil run --max-length` range extended to 16384
6. The `-b` short flag now belongs only to `--batch-size` (not `--backend`)

### Flags that will be added once #67 is resolved:
7. `--validation-split`, `--on-policy-regen-every`, `--log-every`
8. `--warmup-steps`, `--grad-accum`, `--max-grad-norm`, `--mixed-precision`
9. `--checkpoint-dir`, `--save-every`, `--resume-from`
10. `--val-max-new-tokens`

## Proposed solution

Update `docs/cli.md` with:
- A new section for the Unsloth backend
- Updated `distil run` flag table
- Examples for common workflows:
  ```bash
  # Train draft with Unsloth (bf16 draft, 4-bit target)
  sped distil run --backend unsloth --draft-dtype bf16 \
    --draft Qwen/Qwen3-0.6B --target Qwen/Qwen3-8B \
    --dataset ultrachat_200k --max-length 4096

  # Serve with Unsloth 4-bit
  sped serve --backend unsloth --draft Qwen/Qwen3-0.6B --target Qwen/Qwen3-8B
  ```

## Priority

**Low** — should be done after all the CLI flag issues (#67, #77, #78, #79) are resolved, so the docs reflect the final state.

## Files to change
- `docs/cli.md`
